pkg/boot/universalpayload/utilities_test.go

```diff
@@ -0,0 +1,285 @@
+		{
+			name: "testdata/missing_property_load.dtb",
+			fdt: fdtReader(t, &dt.FDT{
+				RootNode: dt.NewNode("/", dt.WithChildren(
+					dt.NewNode("images", dt.WithChildren(
+						dt.NewNode("tianocore", dt.WithProperty(
+							dt.PropertyString("arch", "x86_64"),
+							dt.PropertyString("load", "0x00800000"),
+						)),
+					)),
+				)),
+			}),
+			fdtLoad: nil,
+			err:     fmt.Errorf("failed to convert property '%s' to u64", LoadAddrPropertyName),
+		},
+		// CASE 7: failed to get /images/tianocore/entry-start property
+		{
+			name: "testdata/missing_property_entry_start.dtb",
+			fdt: fdtReader(t, &dt.FDT{
+				RootNode: dt.NewNode("/", dt.WithChildren(
+					dt.NewNode("images", dt.WithChildren(
+						dt.NewNode("tianocore", dt.WithProperty(
+							dt.PropertyString("arch", "x86_64"),
+							dt.PropertyU64("load", 0x00800000),
+						)),
+					)),
+				)),
+			}),
+			fdtLoad: nil,
+			err:     fmt.Errorf("failed to find get '%s' property", EntryAddrPropertyName),
+		},
+		// CASE 8: failed to convert /images/tianocore/entry-start property (type error)
+		{
+			name: "testdata/fail_convert_property_entry_start.dtb",
+			fdt: fdtReader(t, &dt.FDT{
+				RootNode: dt.NewNode("/", dt.WithChildren(
+					dt.NewNode("images", dt.WithChildren(
+						dt.NewNode("tianocore", dt.WithProperty(
+							dt.PropertyString("arch", "x86_64"),
+							dt.PropertyU64("load", 0x00800000),
+							dt.PropertyString("entry-start", "0x00800000"),
+						)),
+					)),
+				)),
+			}),
+			fdtLoad: nil,
+			err:     fmt.Errorf("failed to convert property '%s' to u64", EntryAddrPropertyName),
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := getFdtInfo(tt.name, tt.fdt)
+
+			if tt.err != nil {
+				if tt.err.Error() != err.Error() {
+					t.Errorf("getFdtInfo err = '%v', want '%v'", err, tt.err)
+				}
+				return
+			}
+
+			if err != nil {
+				t.Fatal(err)
+				return
+			}
+
+			if tt.fdtLoad != nil && got == nil {
+				t.Errorf("getFdtInfo fdtLoad = nil, want: %v", tt.fdtLoad)
+				return
+			}
+
+			if tt.fdtLoad != nil {
+				if tt.fdtLoad.Load != got.Load {
+					t.Errorf("getFdtInfo fdtLoad.Load = %d, want %v", got.Load, tt.fdtLoad.Load)
+				}
+				if tt.fdtLoad.EntryStart != got.EntryStart {
+					t.Errorf("getFdtInfo fdtLoad.EntryStart = %v, want %v", got.EntryStart, tt.fdtLoad.EntryStart)
+				}
+			}
+		})
+	}
+}
+
+func mockCPUTempInfoFile(t *testing.T, content string) string {
+	tmpDir := t.TempDir()
+	tempFile, err := os.CreateTemp(tmpDir, "cpuinfo")
+	if err != nil {
+		t.Fatalf("Failed to create temp file: %v", err)
+	}
+
+	sysfsCPUInfoPath = tempFile.Name()
+
+	if _, err := tempFile.WriteString(content); err != nil {
+		t.Fatalf("Failed to write to temp file: %v", err)
+	}
+
+	tempFile.Close()
+	return tempFile.Name()
+}
+
+func TestGetPhysicalAddressSizes(t *testing.T) {
+	tests := []struct {
+		name           string
+		cpuInfoContent string
+		expectedBits   uint8
+		expectedErr    string
```

Comment:
err error, the package should work in such a way that you can use errors.Is, not error strings.